evm-contracts/src/v0.6/dev/AccessController.sol

```diff
@@ -0,0 +1,24 @@
+pragma solidity ^0.6.0;
+
+import "./AccessControllerInterface.sol";
+
+/**
+ * @title Controller
+ * @notice Allows inheriting contracts to control access to functions
+ */
+abstract contract AccessController is AccessControllerInterface {
```

Comment:
Why do we need this contract?

Reply:
It allows for a base contract that all inheriting contracts must create an implementation for `hasAccess` while providing a basic modifier to go along with it. I guess the argument for not needing this contract is that it doesn't have any implemented functions in it already. I'm open to removing it, but it's not necessary for the proxy itself.

Reply:
My objection to it is that it doesn't seem to be used anywhere. 
SimpleAccessControl overrides hasAccess AND checkAccess.

If we want to keep it, checkAccess() should be made non-virtual.